service/src/main/java/com/codeforcommunity/processor/ImportProcessorImpl.java

```diff
@@ -61,4 +70,48 @@ public void importNeighborhoods(
       neighborhood.store();
     }
   }
+
+  @Override
+  public void importReservations(
+      JWTData userData, ImportReservationsRequest importReservationsRequest) {
+    if (userData.getPrivilegeLevel() != PrivilegeLevel.SUPER_ADMIN) {
+      throw new AuthException("User does not have the required privilege level.");
+    }
+
+    // Get superuser ID
+    Integer superAdminId = userData.getUserId();
+
+    for (ReservationImport reservationImport : importReservationsRequest.getReservations()) {
+      if (!db.fetchExists(
+          db.selectFrom(BLOCKS).where(BLOCKS.ID.eq(reservationImport.getBlockId())))) {
+        throw new ResourceDoesNotExistException(reservationImport.getBlockId(), "block");
+      }
+      if (reservationImport.getUserId() != null
+          && (!db.fetchExists(
+              db.selectFrom(USERS).where(USERS.ID.eq(reservationImport.getUserId()))))) {
+        throw new UserDoesNotExistException(reservationImport.getUserId());
```

Comment:
It's weird that we have a special case for users instead of just using the ResourceDoesNotExistException, we might want to simplify and just use the same one for everything. This can be done in a followup PR though

Reply:
Considered it but the reason I left it in for now is that the user exception can take both an ID or username which I liked